### ISSUE TITLE
[makeotfexe] Subroutinization can enter a infinite loop

### ISSUE THREAD

**Issue by benkiel** [adobe-type-tools/afdko#782]
Just had the case with a "bitmap" font with only straight segments, and many short ones. With release mode on, makeotf entered a seemingly (I guess I could have let it run all night, but my patience only went 15 minutes) infinite loop. Know this to be an edge case, but after a certain recursion depth perhaps it should fail? I can see if I can get a test font — these weren't my outlines.

**Comment by miguelsousa** [comment]
This is a **makeotfexe** issue. Its subroutinizer is old and due for replacement (see #331 ).

In the meantime you can use **tx** to subroutinize the CFF table, and **sfntedit** to replace that table using these commands:

```
tx -cff +S +b -std myfont.otf cff_table
sfntedit -a CFF=cff_table myfont.otf
```

**Comment by benkiel** [comment]
Ah! Thanks for that. This can likely be closed then if #331 will fix it.

**Comment by miguelsousa** [comment]
The issues are related, but I see #331 as a "nice to have" whereas this one is "something broken that needs to be fixed". I'll reopen this one since I'd like to investigate if it's possible to patch makeotfexe's current subroutinizer without fully replacing it. The patch would be to cap the level of recursion.

**Comment by miguelsousa** [comment]
@benkiel if you could send me the font offline that would help.

**Comment by benkiel** [comment]
@miguelsousa I'll see if I can get permission

**Comment by benkiel** [comment]
Test file sent to you, @miguelsousa

**Comment by cjchapman** [comment]
@benkiel In theory, this is fixed by #882. In practice, I wasn't able to reproduce the problem with the font you sent us. However, with the new subroutinizer code, makeotf built the font literally 100 times faster. So, I'm going to go ahead and close this issue now. If you still encounter the problem with this new version of the AFDKO that we're about to release (3.0.0),  please let us know. Thanks!

**Comment by benkiel** [comment]
@cjchapman Thank you. I'll test with 3.0.0 when I have the chance (this font has been mastered for the client and shipped).
